### DISCUSSION TITLE
Maestro CI

### DISCUSSION THREAD

**Discussion by GiwrgosMpartsas** [mobile-dev-inc/Maestro#706] [General]
Hey guys, so my question is.. can i run maestro on my own CI? (without using maestros CI) cause in the company i work we have our own.

Thanks

**Comment by thoom76** [comment]
Yes you can. When running the maestro command `maestro test flow.yaml` for example, the maestro command returns exit code 0 when succeeded and non-zero when failed. Including this in your CI results in a succeeding or failing pipeline step :)

**Reply by Jonas1197** [reply]
> Yes you can. When running the maestro command `maestro test flow.yaml` for example, the maestro command returns exit code 0 when succeeded and non-zero when failed. Including this in your CI results in a succeeding or failing pipeline step :)

To be precise while executing actions for example in Github Actions you are not able to provide user input. Good thing is Maestro provides us with the command `maestro --device=[device_name] flow.yaml`. 

Make sure you got a devices connected (be it an emulator or a physical device) beforehand.

Tests will run as they usually do with 'completed' or 'failed' steps as they would appear in the terminal window normally.

**Comment by ETLopes** [comment]
Although I understand that you as a company wants to push Maestro Cloud, I think it would be useful to at least have some code snippets or some guidance on how to run some of these tests on popular CI tools such as Bitrise, CircleCI, github actions.

**Comment by axelniklasson** [comment]
Hi there folks! You can indeed run maestro in your own CI - as you do locally - using the Maestro CLI. If you run into any weird issues with maestro, please file a bug report [here](https://github.com/mobile-dev-inc/maestro/issues/new?assignees=&labels=bug&projects=&template=bug_report.md&title=%5BMaestro+version%2C+e.g.+v1.2.3%5D). Thanks!
